src/Backups/RestoreCoordinationRemote.cpp

```diff
@@ -129,4 +132,49 @@ void RestoreCoordinationRemote::removeAllNodes()
     zk->removeRecursive(zookeeper_path);
 }
 
+bool RestoreCoordinationRemote::hasConcurrentRestores(const String & restore_id, const String & common_restores_path, const std::atomic<size_t> &) const
```

Comment:
Both `common_restores_path` and `restore_id` can be extracted from `zookeeper_path`. Or you can change the constructor of `RestoreCoordinationRemote` and pass them directly from `BackupsWorker`.

Reply:
good point, I have updated constructor